common/ast/src/parser/token.rs

```diff
@@ -788,7 +790,17 @@ impl TokenKind {
             // | TokenKind::UNION
             | TokenKind::WHERE
             // | TokenKind::WINDOW
-            | TokenKind::WITH if !after_as => true,
+            | TokenKind::WITH
+            | TokenKind::DATEADD
+            | TokenKind::YEAR
+            | TokenKind::MONTH
+            | TokenKind::DAY
+            | TokenKind::HOUR
+            | TokenKind::MINUTE
+            | TokenKind::SECOND
+            | TokenKind::DOY
+            | TokenKind::DOW
```

Comment:
If these keywords are not reserved, will there be any ambiguity?

Reply:
I remeber that it will be recognitized as a column.

Reply:
Is the syntax I defined  maybe incorrect？I will check it.

Reply:
If it's recongized as a column, it should be in is_reserved_ident rather than is_reserved_function_name

Reply:
i think reserving only the date_add as function namd should solve the problem.

Reply:
Ok